docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
+
+You can also see the [full changes in PR #18929](https://github.com/gatsbyjs/gatsby/pull/18928/files) for an overview of the changes you have to make to get MDX working.
+
+### Step 1
+
+Install [gatsby-plugin-mdx](/packages/gatsby-plugin-mdx/), the official plugin for using MDX with Gatsby. Also, install `@mdx-js/mdx` and `@mdx-js/react`.
+
+```
+npm install --save gatsby-plugin-mdx @mdx-js/mdx @mdx-js/react
+```
+
+### Step 2
+
+In `gatsby-config.js`, replace `gatsby-transformer-remark` with `gatsby-plugin-mdx`, along with the following changes:
+
+```diff:title=gatsby-config.js
+  {
+-  resolve: `gatsby-transformer-remark`,
++  resolve: `gatsby-plugin-mdx`,
+   options: {
++    extensions: [`.mdx`, `.md`],
+-    plugins: [
++    gatsbyRemarkPlugins: [
+       {
+         resolve: `gatsby-remark-images`,
+         options: {
+           maxWidth: 590,
+         },
+       },
+       {
+         resolve: `gatsby-remark-responsive-iframe`,
+         options: {
+           wrapperStyle: `margin-bottom: 1.0725rem`,
+         },
+       },
+       `gatsby-remark-prismjs`,
+       `gatsby-remark-copy-linked-files`,
+       `gatsby-remark-smartypants`,
+     ],
+   },
+  },
+```
+
+Then, add some options to `gatsby-plugin-feed`. This will allow the the RSS feed of the site to parse MDX.
+
+```diff:title=gatsby-config.js
+- `gatsby-plugin-feed`,
+
++ {
++   resolve: `gatsby-plugin-feed`,
++   options: {
++     query: `
++       {
++         site {
++           siteMetadata {
++             title
++             description
++             siteUrl
++           }
++         }
++       }
++     `,
++     feeds: [
++       {
++         serialize: ({ query: { site, allMdx } }) =>
++           allMdx.edges.map(edge =>
++             Object.assign({}, edge.node.frontmatter, {
++               description: edge.node.excerpt,
++               data: edge.node.frontmatter.date,
++               url: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               guid: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               custom_elements: [{ "content:encoded": edge.node.body }],
++             })
++           ),
++         query: `
++         {
++           allMdx(
++             limit: 1000,
++             sort: { order: DESC, fields: [frontmatter___date] },
++           ) {
++             edges {
++               node {
++                 fields { slug }
++                 frontmatter {
++                   title
++                   date
++                 }
++                 body
++               }
++             }
++           }
++         }
++         `,
++         output: `/rss.xml`,
++         title: `Gatsby RSS feed`,
++       },
++     ],
++   },
++ },
+```
```

Comment:
There is a lot of information here. Is it worth a brief explanation or link that explains what's happening?

Reply:
there was a suggestion to place this in a new plugin `gatsby-plugin-feed-mdx` - i can not find this suggestion anymore :(

but maybe then it is not needed to post so much informations here ...

Reply:
found the suggestion here: https://github.com/gatsbyjs/gatsby/pull/18928#discussion_r344594507

Reply:
Yeah that plugin would be helpful!

Reply:
Definitely something to consider, but for the moment let's see what we can do with the existing configuration. Is there some place we can call out for users who want more context or information?

Reply:
@laurieontech @muescha replaced the big diff with one line via `gatsby-plugin-feed-mdx` - https://github.com/thomaswangio/gatsby-plugin-feed-mdx 😃

Tested in https://github.com/thomaswangio/gatsby-personal-starter-blog/commit/4daab2d1609c560b62a9580e9d4fe16b878c4fe8 and it worked fine!